src/form/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {
+  ChangeEvent,
+  FC,
+  ReactElement,
+  useCallback,
+  useEffect,
+  useRef,
+  useState
+} from 'react';
+import { format as formatDate, isValid, parse } from 'date-fns';
+
+import { IconButton } from '@/elements';
+import { Menu } from '@/layers';
+import { Card } from '@/layout';
+import { Placement } from '@/utils';
+import { Calendar, Input, InputProps, InputRef } from '@/form';
+
+import CalendarIcon from '@/assets/icons/calendar.svg?react';
+
+export type DateInputProps = Omit<InputProps, 'value' | 'onChange'> & {
+  /**
+   * The format in which the date should be displayed.
+   * @type {string}
+   */
+  format?: string;
+
+  /**
+   * Calendar placement type.
+   */
+  placement?: Placement;
+
+  /**
+   * Icon to show in open calendar button.
+   */
+  icon?: ReactElement;
+} & (
```

Comment:
this doesn't look right

Reply:
It's for correct types:
<img width="531" alt="Screenshot 2024-05-24 at 15 37 28" src="https://github.com/reaviz/reablocks/assets/1290843/05afb24f-2c80-49d8-96d9-abc2e3966c79">
<img width="758" alt="Screenshot 2024-05-24 at 15 37 39" src="https://github.com/reaviz/reablocks/assets/1290843/6fbd1598-191d-4562-99db-8d25742bfab5">